implementations/micrometer-registry-new-relic/src/main/java/io/micrometer/newrelic/NewRelicConfig.java

```diff
@@ -53,6 +53,16 @@ default String eventType() {
         return v;
     }
 
+    /**
+     * When this is {@code true}, the New Relic metrics will be published with the NewRelicAgentClientProvider.
+     * Defaults to {@code false}.
+     * @return whether to use the NewRelicAgentClientProvider
+     */
+    default boolean agentClientProviderEnabled() {
```

Comment:
I'm wondering if an Enum wouldn't be better than a boolean. If we want to integrate the now [separate New Relic registry](https://github.com/newrelic/micrometer-registry-newrelic) into this repository, it might make sense as just another client provider implementation.

Reply:
Good point. That would be more flexible.  Let me update it.

Reply:
Given the potential to add support for New Relic's implementation I was explicit with the enum values.  The original integration is with their Insights API.  Their new registry would makes sense as a "metrics_api" value.

Reply:
Done with the change.